src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -389,14 +413,17 @@ function read_global_30s_landuse(path) result(iErr)
       integer, parameter :: tile_x = 1200       ! x-dimension of each tile of global 30-arc-second landuse
       integer, parameter :: tile_y = 1200       ! y-dimension of each tile of global 30-arc-second landuse
 
-      integer :: istatus
+      integer (c_int) :: istatus
       integer :: ix, iy
-      integer :: isigned, endian, wordsize, nx, ny, nz
-      real (kind=R4KIND) :: scalefactor
-      real (kind=R4KIND), dimension(:,:,:), allocatable :: tile
+      integer (c_int) :: isigned, endian, wordsize, nx, ny, nz
+      real (c_float) :: scalefactor
+      real (c_float), dimension(:,:,:), pointer, contiguous :: tile
+      type (c_ptr) :: tile_ptr
       character(len=StrKIND) :: filename
+      character(len=StrKIND+1, kind=c_char) :: c_filename
```

Comment:
Let's change the definition of `c_filename` here to match what was done in the `read_global_30s_topo` routine.